app/models/manageiq/providers/amazon/cloud_manager/scanning/job.rb

```diff
@@ -0,0 +1,46 @@
+class ManageIQ::Providers::Amazon::CloudManager::Scanning::Job < VmScan
+  # Make updates to default state machine for scanning on AWS
+  #
+  ######################################################################################################################################################
+  #
+  # AWS SmartState flow:
+  #
+  #   - vm.scan_metadata is called from VmScan#call_scan method
+  #     * vm.scan_metadata method lives in ScanningMixin
+  #   - vm.scan_metadata calls MiqServer::ServerSmartProxy#queue_call
+  #   - MiqServer::ServerSmartProxy#queue_call passes control (via MiqQueue) to MiqServer::ServerSmartProxy#scan_metadata
+  #   - MiqServer::ServerSmartProxy#scan_metadata calls ManageIQ::Providers::Amazon::CloudManager::VmOrTemplateShared::Scanning#perform_metadata_scan
+  #   - The perform_metadata_scan method queues the request via Amazon's SQS
+  #     * Then the job is marked as finished - and THAT is why the transition below is needed
+  #
+  #   - In a different worker process, ManageIQ::Providers::Amazon::AgentCoordinatorWorker::Runner
+  #     * checks if a ManageIQ agent is running in the Amazon AWS Cloud, and starts it, if needed
```

Comment:
@roliveri @hsong-rh @jerryk55 @Fryguy why not leave the job in the `scanning` state here?  Then, in ManageIQ::Providers::Amazon::AgentCoordinatorWorker::Runner::ResponseThread#perform_metadata_sync, move to `synchronizing` state before going to `:finished` state?